pkg/virt-controller/services/template.go

```diff
@@ -878,32 +878,34 @@ func (t *templateService) renderLaunchManifest(vmi *v1.VirtualMachineInstance, i
 	resources.Requests = make(k8sv1.ResourceList)
 	resources.Limits = make(k8sv1.ResourceList)
 
+	// Copy vmi resources requests to a container
+	for key, value := range vmiResources.Requests {
+		resources.Requests[key] = value
+	}
+
+	// Copy vmi resources limits to a container
+	for key, value := range vmiResources.Limits {
+		resources.Limits[key] = value
+	}
+
 	// Set Default CPUs request
 	if !vmi.IsCPUDedicated() {
 		vcpus := int64(1)
 		if vmi.Spec.Domain.CPU != nil {
 			vcpus = hardware.GetNumberOfVCPUs(vmi.Spec.Domain.CPU)
 		}
 		cpuAllocationRatio := t.clusterConfig.GetCPUAllocationRatio()
-		if vcpus != 0 && cpuAllocationRatio > 0 {
+		if vcpus != 0 && cpuAllocationRatio > 1 {
 			val := float64(vcpus) / float64(cpuAllocationRatio)
 			vcpusStr := fmt.Sprintf("%g", val)
 			if val < 1 {
 				val *= 1000
 				vcpusStr = fmt.Sprintf("%gm", val)
 			}
 			resources.Requests[k8sv1.ResourceCPU] = resource.MustParse(vcpusStr)
+			resources.Limits[k8sv1.ResourceCPU] = *resource.NewQuantity(vcpus, resource.BinarySI)
```

Comment:
This is interesting. Would you be able to extract this into a separate commit and provide a reason?

Reply:
Setting a limit, as long as the limit is greater than the request, retains the Burstable QoS, right? https://kubernetes.io/docs/tasks/configure-pod-container/quality-service-pod/#create-a-pod-that-gets-assigned-a-qos-class-of-guaranteed